docs/fsharp/style-guide/formatting.md

```diff
@@ -0,0 +1,505 @@
+
+F# uses significant whitespace by default. The following guidelines are intended to provide guidance as to how to juggle some challenges this can impose.
+
+### Using spaces
+
+When indentation is required, you must use spaces, not tabs. At least one space is required. Your organization can create coding standards to specify the number of spaces to use for indentation; two, three or four spaces of indentation at each level where indentation occurs is typical.
+
+**We recommend 4 spaces per indentation.**
+
+That said, indentation of programs is a subjective matter. Variations are OK, but the first rule you should follow is *consistency of indentation*. Choose a generally accepted style of indentation and use it systematically throughout your codebase.
+
+## Formatting discriminated union declarations
+
+Indent `|` in type definition by 4 spaces:
+
+```fsharp
+// OK
+type Volume =
+    | Liter of float
+    | FluidOunce of float
+    | ImperialPint of float
+
+// Not OK
+type Volume =
+| Liter of float
+| USPint of float
+| ImperialPint of float
+```
+
+Instantiated Discriminated Unions that split across multiple lines should give contained data a new scope with indentation:
+
+```fsharp
+let tree1 =
+    BinaryNode
+        (BinaryNode(BinaryValue 1, BinaryValue 2),
+         BinaryNode(BinaryValue 3, BinaryValue 4))
+```
+
+The closing parenthesis can also be on a new line:
+
+```fsharp
+let tree1 =
+    BinaryNode(
+        BinaryNode(BinaryValue 1, BinaryValue 2),
+        BinaryNode(BinaryValue 3, BinaryValue 4)
+    )
+```
+
+## Formatting tuples
+
+A tuple instantiation should be parenthesized, and the delimiting commas within should be followed by a single space, for example: `(1, 2)`, `(x, y, z)`.
+
+A commonly accepted exception is to omit parentheses in pattern matching of tuples:
+
+```fsharp
+let (x, y) = z // Destructuring
+
+match x, y with
+| 1, _ -> 0
+| x, 1 -> 0
+| x, y -> 1
+```
+
+## Formatting records
+
+Short records can be written in one line:
+
+```fsharp
+let point = { X = 1.0; Y = 0.0 }
+```
+
+Records that are longer should use new lines for labels:
+
+```fsharp
+let rainbow =
+    { Boss = "Jeffrey"
+      Lackeys = ["Zippy"; "George"; "Bungle"] }
+```
+
+Placing the opening token on the same line and the closing token on a new line is also fine:
+
+```fsharp
+let rainbow = {
+    Boss1 = "Jeffrey"
+    Boss2 = "Jeffrey"
+    Boss3 = "Jeffrey"
+    Boss4 = "Jeffrey"
+    Boss5 = "Jeffrey"
+    Boss6 = "Jeffrey"
+    Boss7 = "Jeffrey"
+    Boss8 = "Jeffrey"
+    Lackeys = ["Zippy"; "George"; "Bungle"]
+}
+```
+
+The same rules apply for list and array elements.
+
+## Formatting lists and arrays
+
+Write `x :: l` with spaces around the `::` operator (`::` is an infix operator, hence surrounded by spaces) and `[1; 2; 3]` (`;` is a delimiter, hence followed by a space).
+
+Always use at least one space between two distinct brace-like operators (e.g. leave a space between a `[` and a `{`).
```

Comment:
e.g. -> for example

better to start a new sentence instead of putting the sentence in parentheses.